src/miscutils/logging_config.py

```diff
@@ -68,37 +93,43 @@ def default_logging(
             ],
         )
     elif log_format == LogFormat.JSON:
-        # Setup json logging.
-        logging.config.dictConfig(
-            {
-                "version": 1,
-                "formatters": {
-                    "json": {
-                        "format": "%(message)s $(lineno)d $(filename)s",
-                        "class": "pythonjsonlogger.jsonlogger.JsonFormatter",
-                    }
-                },
-                "handlers": {"json": {"class": "logging.StreamHandler", "formatter": "json"}},
-                "loggers": {"": {"handlers": ["json"], "level": level}},
-            }
-        )
-
+        pre_chain = [
+            add_line_no,
+            add_file_name,
+            structlog.stdlib.add_logger_name,
+            structlog.stdlib.add_log_level,
+            structlog.stdlib.PositionalArgumentsFormatter(),
+            structlog.processors.StackInfoRenderer(),
+            structlog.processors.format_exc_info,
+            structlog.processors.UnicodeDecoder(),
+        ]
         structlog.configure(
             context_class=structlog.threadlocal.wrap_dict(dict),
             logger_factory=structlog.stdlib.LoggerFactory(),
             wrapper_class=structlog.stdlib.BoundLogger,
             cache_logger_on_first_use=True,
-            processors=[
-                structlog.stdlib.filter_by_level,
-                structlog.stdlib.add_logger_name,
-                structlog.stdlib.add_log_level,
-                structlog.stdlib.PositionalArgumentsFormatter(),
-                structlog.processors.StackInfoRenderer(),
-                structlog.processors.format_exc_info,
-                structlog.processors.UnicodeDecoder(),
-                structlog.stdlib.render_to_log_kwargs,
-            ],
+            processors=[structlog.stdlib.filter_by_level]
+            + pre_chain
+            + [structlog.stdlib.ProcessorFormatter.wrap_for_formatter],
+        )
+
+        formatter = structlog.stdlib.ProcessorFormatter(
+            processor=structlog.processors.JSONRenderer(), foreign_pre_chain=pre_chain
         )
+        handler = logging.StreamHandler()
+        handler.setFormatter(formatter)
+        root_logger = logging.getLogger()
+        root_logger.handlers.clear()
+        root_logger.addHandler(handler)
+        root_logger.setLevel(level)
+
+        # Some loggers have existing handlers that need to be overridden
+        if override_logs:
```

Comment:
Need to do this explicitly for `"uvicorn.access"`.  A bit weird since configuring the root logger alone works for the other uvicorn loggers.